### PR TITLE
add GHA workflows

### PR DESCRIPTION
**Pull request by Xiangs18**

### CODE REVIEW

File: .github/codeql.yml
Comment (MrCreosote) [inline .github/codeql.yml:1]: I assume all of these workflows are the standard versions?

File: codeql.yml
Comment (MrCreosote) [inline codeql.yml:1]: why is this file and dependabot duplicated?

File: .github/codeql.yml
Comment (Xiangs18) [reply .github/codeql.yml:1]: yeah

File: codeql.yml
Comment (Xiangs18) [reply codeql.yml:1]: deleted

**Review by MrCreosote — COMMENTED** [review]
LGTM, but @briehl should give final approval

File: .github/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+jobs:
+  CodeQL-Build:
+
+    # CodeQL runs on ubuntu-latest and windows-latest
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
```
Comment (bio-boris) [inline .github/codeql.yml:17]: This one can be updated too

Comment (Xiangs18) [reply .github/codeql.yml]: 👍

File: .github/codeql.yml
Comment (MrCreosote) [reply .github/codeql.yml:1]: yep, whatever we're using for the other repos

**Review by briehl — COMMENTED** [review]
I just have one detail to ask about - dependency update rate.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+- package-ecosystem: docker
+  directory: "/"
+  schedule:
+    interval: weekly
+    time: '11:00'
+  open-pull-requests-limit: 10
+- package-ecosystem: pip
+  directory: "/"
+  schedule:
+    interval: daily
```
[the remark this answers is not in the document]
Comment (bio-boris) [reply .github/dependabot.yml:12]: We don't have any recs atm

[the remark this answers is not in the document]
Comment (briehl) [reply .github/dependabot.yml:12]: Well, IMO daily is still overkill for a service that isn't being actively worked on or deployed. I'd go with a monthly maintenance schedule to keep the dependabot spam to a minimum. It's not like it has a ton of dependencies.

[the remark this answers is not in the document]
Comment (bio-boris) [reply .github/dependabot.yml:12]: Sounds good to me.

[the remark this answers is not in the document]
Comment (Xiangs18) [reply .github/dependabot.yml]: I changed intervals to weekly, a balanced approach.

**Review by briehl — APPROVED** [review]

**Pull request merged**